File: appgal-cloud/src/main/java/io/appgal/cloud/model/DropOffNotification.java
package io.appgal.cloud.model;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.Serializable;

public class DropOffNotification implements Serializable {
    private static Logger logger = LoggerFactory.getLogger(DropOffNotification.class);

    private SourceOrg sourceOrg;
    private Location location;
    private FoodRunner foodRunner;

    public DropOffNotification() {
    }

    public DropOffNotification(SourceOrg sourceOrg, Location location, FoodRunner foodRunner) {
        this.sourceOrg = sourceOrg;
        this.location = location;
        this.foodRunner = foodRunner;
    }

    public SourceOrg getSourceOrg() {
        return sourceOrg;
    }

    public void setSourceOrg(SourceOrg sourceOrg) {
        this.sourceOrg = sourceOrg;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public FoodRunner getFoodRunner() {
        return foodRunner;
    }

    public void setFoodRunner(FoodRunner foodRunner) {
        this.foodRunner = foodRunner;
    }

    public static DropOffNotification parse(String json)
    {
        DropOffNotification dropOffNotification = new DropOffNotification();

        JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
        dropOffNotification.sourceOrg = SourceOrg.parse(jsonObject.get("sourceOrg").toString());
        dropOffNotification.location = Location.parse(jsonObject.get("location").toString());
        dropOffNotification.foodRunner = FoodRunner.parse(jsonObject.get("foodRunner").toString());

        return dropOffNotification;
    }

    public JsonObject toJson()
    {
        JsonObject jsonObject = new JsonObject();
        jsonObject.add("foodRunner", this.foodRunner.toJson());
        jsonObject.add("sourceOrg", this.sourceOrg.toJson());
        jsonObject.add("location", this.location.toJson());
        return jsonObject;
    }

    @Override
    public String toString()
    {
        JsonObject jsonObject = new JsonObject();

        jsonObject.add("sourceOrg", this.sourceOrg.toJson());
        jsonObject.add("location", this.location.toJson());
        jsonObject.add("foodRunner", this.foodRunner.toJson());

        return jsonObject.toString();
    }
}

File: appgal_foodrunner_app/app/lib/src/model/dropOffNotification.dart
import 'dart:convert';

import 'package:app/src/model/foodRunner.dart';
import 'package:app/src/model/sourceOrg.dart';

class DropOffNotification
{
  SourceOrg sourceOrg;
  FoodRunner foodRunner;

  DropOffNotification(SourceOrg sourceOrg, FoodRunner foodRunner)
  {
    this.sourceOrg = sourceOrg;
    this.foodRunner = foodRunner;
  }

  DropOffNotification.fromJson(Map<String, dynamic> json)
  { 
    if(json["sourceOrg"] != null)
    {
      this.sourceOrg = SourceOrg.fromJson(json["sourceOrg"]);
    }
    if(json["foodRunner"] != null)
    {
      this.foodRunner = FoodRunner.fromJson(json["foodRunner"]);
    }
  }

  Map<String, dynamic> toJson() 
  {
    Map<String, dynamic> map = new Map();
    if(this.sourceOrg != null)
    {
      map["sourceOrg"] = this.sourceOrg;
    }
    if(this.sourceOrg != null)
    {
      map["foodRunner"] = this.foodRunner;
    }
    return map;
  }

  String toString()
  {
    String json = jsonEncode(this.toJson());
    return json;
  }
}